game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1067,22 +1059,25 @@ private void pushFightLoopOnStack(final boolean firstRun) {
    * still needed. They can be safely removed once save compatibility can be broken.
    */
   @VisibleForTesting
-  public List<IExecutable> getBattleExecutables(final boolean firstRun) {
+  public List<IExecutable> getBattleExecutables() {
     final List<IExecutable> steps = new ArrayList<>();
-    addFightStartSteps(firstRun, steps);
+    addFightStartSteps(steps);
     addFightSteps(steps);
     addCheckEndBattleAndRetreatingSteps(steps);
     return steps;
   }
 
-  private void addFightStartSteps(final boolean firstRun, final List<IExecutable> steps) {
+  private void addFightStartSteps(final List<IExecutable> steps) {
```

Comment:
Codacy found an issue: [Avoid really long methods.](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=5791556)

Reply:
That function is pre-existing.  And most of the function now is the anonymous inner class definitions that I can't move yet because it would break save game compatibility.

Reply:
I'll convert the last unconverted step in that method and then I should be able to move the class definitions outside of the method body.